Classes/Security/FilePermissionsAspect.php

```diff
@@ -57,12 +71,11 @@ protected function respectStorage(Matcher $matcher) {
 	}
 
 	/**
-	 * @param Matcher $matcher
+	 * @param Query $query
+	 * @param ConstraintInterface|NULL $constraints
 	 * @return void
 	 */
-	protected function respectFileMounts(Matcher $matcher) {
-		$matcher->setLogicalSeparatorForLike(Matcher::LOGICAL_OR);
-
+	protected function respectFileMounts(Query &$query, &$constraints) {
```

Comment:
Testing the patch and it looks it works fine for me without the reference annotation "&". If it doesn't for your - after double checking - I would consider a return of the object constraints. The return of the slot function must be the same as the signature of the signal, if I recall correctly, e.g `return array($query, $constraints)`

Also my version of PHP if it influences in any ways:

```
php -v                                                                                                                                                                                          
PHP 5.6.4 (cli) (built: Jan  7 2015 11:53:05)
```